FILPATHLEN became HAMLIB_FILPATHLEN in hamlib 4.2

FILPATHLEN was changed to HAMLIB_FILPATHLEN starting in [hamlib 4.2](https://github.com/Hamlib/Hamlib/blob/ff9a745c7a716791d58932ed6b82036823094ab7/NEWS#L40), so tlf does not currently build against hamlib 4.2. The error message as expected is `sendqrg.c:103:6: error: ‘FILPATHLEN’ undeclared (first use in this function)`. Presumably the fix is to update `sendqrg.c` to use the new macro. I don't see any pre-processor conditions elsewhere in the code, but one might need to be introduced here to maintain compatibility with older hamlib versions as well.

--- a/src/sendqrg.c
+++ b/src/sendqrg.c
@@ -87,7 +87,7 @@ int init_tlf_rig(void) {
 
     rigportname[strlen(rigportname) - 1] = '\0';	// remove '\n'
     strncpy(my_rig->state.rigport.pathname, rigportname,
-	    FILPATHLEN - 1);
+	    TLFFILPATHLEN - 1);
 
     caps = my_rig->caps;
 
@@ -104,9 +104,9 @@ int init_tlf_rig(void) {
     if (dcd_type != RIG_DCD_NONE)
 	my_rig->state.dcdport.type.dcd = dcd_type;
     if (ptt_file)
-	strncpy(my_rig->state.pttport.pathname, ptt_file, FILPATHLEN);
+	strncpy(my_rig->state.pttport.pathname, ptt_file, TLFFILPATHLEN);
     if (dcd_file)
-	strncpy(my_rig->state.dcdport.pathname, dcd_file, FILPATHLEN);
+	strncpy(my_rig->state.dcdport.pathname, dcd_file, TLFFILPATHLEN);
 
     my_rig->state.rigport.parm.serial.rate = serial_rate;
 

--- a/src/sendqrg.h
+++ b/src/sendqrg.h
@@ -28,6 +28,8 @@
 #else
   #ifdef FILPATHLEN
   #define TLFFILPATHLEN FILPATHLEN
+  #else
+  #error "FILPATHLEN macro not found"
   #endif
 #endif
 

--- a/src/sendqrg.h
+++ b/src/sendqrg.h
@@ -23,6 +23,14 @@
 
 #include <hamlib/rig.h>
 
+#ifdef HAMLIB_FILPATHLEN
+  #define TLFFILPATHLEN HAMLIB_FILPATHLEN
+#else
+  #ifdef FILPATHLEN
+  #define TLFFILPATHLEN FILPATHLEN
+  #endif
+#endif
+
 int init_tlf_rig(void);
 void close_tlf_rig(RIG *my_rig);
 

--- a/src/sendqrg.h
+++ b/src/sendqrg.h
@@ -29,7 +29,7 @@
   #ifdef FILPATHLEN
   #define TLFFILPATHLEN FILPATHLEN
   #else
-  #error "FILPATHLEN macro not found"
+  #error "(HAMLIB_)FILPATHLEN macro not found"
   #endif
 #endif
 

